sockeye/vocab.py

```diff
@@ -32,23 +32,41 @@
 InverseVocab = Dict[int, str]
 
 
-def build_from_paths(paths: List[str], num_words: Optional[int] = None, min_count: int = 1,
-                     pad_to_multiple_of: Optional[int] = None) -> Vocab:
+def build_from_paths(paths: Union[List[str], str]) -> Counter:
     """
-    Creates vocabulary from paths to a file in sentence-per-line format. A sentence is just a whitespace delimited
-    list of tokens. Note that special symbols like the beginning of sentence (BOS) symbol will be added to the
-    vocabulary.
-
     :param paths: List of paths to files with one sentence per line.
+    :return: Token counter.
+    """
+    with ExitStack() as stack:
+        logger.info("Building vocabulary from dataset(s): %s", paths)
+        if isinstance(paths, List):
+            files = (stack.enter_context(utils.smart_open(path, mode='rt')) for path in paths)
+        else:
+            files = stack.enter_context(utils.smart_open(paths, mode='rt'))
+        return count_tokens(chain(*files))
+
+def build_from_shards(paths: Iterable[Union[Tuple[str], str]], num_words: Optional[int] = None, min_count: int = 1,
```

Comment:
I'd maybe change from `Tuple` to `Sequence` here and above (which we'd need to import still). For Tuple we'd need the ellipsis.

```suggestion
def build_from_shards(paths: Iterable[Union[Sequence[str], str]], num_words: Optional[int] = None, min_count: int = 1,
```